### PR TITLE
feat(tax bracket): tax bracket service

### PR DESCRIPTION
**Pull request by kvhswong**
- Closes: HOME-549

### CODE REVIEW

**Comment by kvhswong** [comment]
Still need to get it into the build process. But pushed so I can get reviews.

**Comment by kvhswong** [comment]
Was told by one of my Go friends that, https://onsi.github.io/gomega/, is a good matcher testing library.

**Comment by zrangerz** [comment]
You're modeling the new thing you tell us! Want to see on Monday!

--
Steve Mays
Sent with Airmail

On August 19, 2016 at 3:25:03 PM, Kevin Wong (notifications@github.com<mailto:notifications@github.com>) wrote:

Was told by one of my Go friends that, https://onsi.github.io/gomega/, is a good matcher testing library.

**Comment by gregdurham** [comment]
This looks good, granted my knowledge is now likely less than yours. I think we should add a further piece to add [go-metrics](https://github.com/rcrowley/go-metrics) to collect information just as a general framework. We should also do something standard for logging like you did for the client-billing

**Comment by gregdurham** [comment]
I would look at govendor possible to manage dependencies, i have seen this used by hashicorp for their projects

**Comment by kvhswong** [comment]
Looks like we reached the 50 repo limit for our GitHub license. Time to upgrade or pare down repositories. For now the tax-bracket repo is public.

screen shot 2016-08-19 at 11 02 30 pm

**Comment by zrangerz** [comment]
Let's chat Monday and delete some first.

--
Steve Mays
Sent with Airmail

On August 19, 2016 at 4:03:43 PM, Kevin Wong (notifications@github.com<mailto:notifications@github.com>) wrote:

Looks like we reached the 50 repo limit for our GitHub license. Time to upgrade or pare down repositories. For now the tax-bracket repo is public.

[screen shot 2016-08-19 at 11 02 30 pm]<https://cloud.githubusercontent.com/assets/17414422/17826538/2403da76-6661-11e6-8667-f90f7b0e42b6.png>

File: README.md [hunk 1]
```diff
@@ -1 +1,31 @@
...
+
+## Test
+
+    go test
+
+## Health Check
+
+Health of the service can be queried at the endpoint /taxstatus/health. If the service is healthy, a 200 http response code is returned. If the service is not up a 503 is returned. The health of the service is dependent on whether the current tax year brackets can be loaded. If they cannot then an http 503 code is returned.
+
+    curl http://localhost:8085/taxstatus/health
```
Comment (abuecker) [inline README.md]: Probably want this:

     curl http://localhost:8085/taxbracket/health

**Comment by bondricek** [comment]
go is new to me but overall looks good.
i would suggest adding a few more things to the readme,   examples of the other 2 routes and maybe the list of filing statues.  might also be worth adding a comment that the calc tests will be needed to be updated every year.

**Comment by gregdurham** [comment]
Do we have API documentation for this service, this way we can possibly use it to test this service and make sure it works to the 'contract'

**Comment by kvhswong** [comment]
No API docs at the moment. Could put a docs project together similar to advisor or consumer.

**Comment by gregdurham** [comment]
may be a good idea if for nothing else but documentation

**Comment by bondricek** [comment]
could include raml in the same project in a sub directory, would be easier to keep in sync then a separate project

**Comment by zrangerz** [comment]
Id be good with just examples in README.md for now as this is internally facing not external.  Lets ship it, write improvements, do improvements.

Steve

--
Steve Mays
Sent with Airmail

**Comment by gregdurham** [comment]
@bondricek i agree with keeping it with the project

**Comment by zrangerz** [comment]
Improve logging in another story. Thanks for the examples in the readme.

+1

**Pull request merged**